agent/exec/dockerapi/executor.go

```diff
@@ -17,15 +17,23 @@ type executor struct {
 	client           engineapi.APIClient
 	secrets          exec.SecretsManager
 	genericResources []*api.GenericResource
+	node             *api.NodeDescription
 }
 
 // NewExecutor returns an executor from the docker client.
 func NewExecutor(client engineapi.APIClient, genericResources []*api.GenericResource) exec.Executor {
-	return &executor{
+	var executor = &executor{
 		client:           client,
 		secrets:          secrets.NewManager(),
 		genericResources: genericResources,
 	}
+	// Get the node information from the newly created executor
+	var ctx = context.Background()
```

Comment:
I added a mutex to the executor field. Sorry but it is my first time using Golang. I hope the code makes sense.